src/lib/logger.ts

```diff
@@ -52,13 +52,23 @@ export const log = {
             properties: additionalProperties,
         });
     },
-    trackEvent: (eventName: string) => {
-        browser &&
+    trackEvent: (eventName: string, dimensions: string | undefined) => {
+        if (dimensions && browser) {
             appInsights.trackEvent({
                 name: eventName,
                 properties: {
                     ...additionalProperties,
+                    dimensions,
                 },
             });
+        } else {
+            browser &&
+                appInsights.trackEvent({
+                    name: eventName,
+                    properties: {
+                        ...additionalProperties,
+                    },
+                });
```

Comment:
@jwinston-bn, @MrGrinst, I debated this block for a while. Personally, I don't love it and am open to suggestions. I was trying to avoid having:
```
properties: {
    ...
    dimensions: undefined,
},
```
for almost every other app insight event. But TypeScript being TypeScript, it becomes a pain.

Reply:
I've seen it done like this before:
```
            appInsights.trackEvent({
                name: eventName,
                properties: {
                    ...additionalProperties,
                    ...(dimensions && { dimensions }),
                },
            });
```
since spreading `undefined` won't affect the object. Looks weirder but less code, so there's pros and cons I guess.

Reply:
Is this going to just have a comma separated list in the single property though? Ideally we'd convert this to key-value pairs. That way it'll be an actual object in the app insights data within custom dimensions and we can search on it.